### PR TITLE
Add Stork to /articles

### PR DESCRIPTION
**Pull request by healeycodes**
Let's take Search out of beta and move it to `/articles`.

If a user has JavaScript turned off, make sure we don't show an empty block.

Also, bump Stork version.

### CODE REVIEW

**Comment by healeycodes** [comment]
Hey @jameslittle230 what do you think of this? Any design improvements you think I could make here?

See preview link in the PR comments here ☝️

File: pages/articles.tsx [hunk 1]
```diff
@@ -28,6 +33,26 @@ export default function Articles({ tags, posts }) {
             ))
             .reduce((prev, curr) => [prev, ", ", curr])}
         </p>
+        <div className="stork">
```
Comment (jameslittle230) [inline pages/articles.tsx:36]: I think if you wanted, you can apply all this to the inner `stork-wrapper` div. That div doesn't do anything special (it's just for the theme), so you can hide/show it all you want!

File: pages/articles.tsx [hunk 2]
```diff
@@ -40,7 +65,20 @@ export default function Articles({ tags, posts }) {
...
       `}</style>
+
+      {/* Ensure we don't show empty space for users without JS */}
+      <noscript>
+        <style>{`
+          .stork {
+            display: none;
+          }`}
+        </style>
+      </noscript>
```
Comment (jameslittle230) [inline pages/articles.tsx:81]: This is smart!!

File: pages/articles.tsx [hunk 3]
```diff
@@ -16,6 +18,9 @@ export default function Articles({ tags, posts }) {
   );
   return (
     <Layout title={title} description="A list of every article I've written.">
+      <Head>
+        <link rel="stylesheet" href="https://files.stork-search.net/basic.css" />
```
Comment (jameslittle230) [inline pages/articles.tsx:22]: `/flat.css` might be better aligned with your site's design! Up to you!

File: pages/articles.tsx [hunk 4]
```diff
@@ -28,6 +33,26 @@ export default function Articles({ tags, posts }) {
...
+        <div className="stork">
+          <div className="stork-wrapper">
+            <input placeholder="Search all posts.." data-stork="posts" className="stork-input" />
+            <div data-stork="posts-output" className="stork-output"></div>
+          </div>
+        </div>
+        <Script
+          src="https://files.stork-search.net/stork.js"
+          onLoad={() => {
+            // @ts-ignore
```
Comment (jameslittle230) [inline pages/articles.tsx:45]: Curious - what does TS complain about here?

**Review by jameslittle230 — APPROVED** [review]
I really like the implementation!

My only question (which is outside the scope of this PR regardless) is if it's possible to untie the execution of `/lib/stork.js` from index.js's `getStaticProps` - it probably wouldn't matter for you, but it'd be nice to find a solution that runs after each static page is rendered, and indexes the HTML instead of the markdown files on disk.

File: pages/articles.tsx (hunk 3 above)
Comment (healeycodes) [reply pages/articles.tsx]: Awesome! Fixed.

File: pages/articles.tsx (hunk 1 above)
Comment (healeycodes) [reply pages/articles.tsx]: Neat idea, fixed.

File: pages/articles.tsx (hunk 4 above)
Comment (healeycodes) [reply pages/articles.tsx:43]: `Property 'stork' does not exist on type 'Window & typeof globalThis'.ts(2339)`

It didn't like my attempted hack of:

```ts
window as Window & { stork: { register: (a, b) => {}} }
```

I think the solution is a stork library + types (and not loading it via `<script>`).

**Comment by healeycodes** [comment]
> I really like the implementation!
> 
> My only question (which is outside the scope of this PR regardless) is if it's possible to untie the execution of `/lib/stork.js` from index.js's `getStaticProps` - it probably wouldn't matter for you, but it'd be nice to find a solution that runs after each static page is rendered, and indexes the HTML instead of the markdown files on disk.

@jameslittle230 this would be my ideal solution as well. Off the top of my head, and after a brief Google search, I'm not sure if there is a 'post-build' step for Next.js 🤔

**Comment by healeycodes** [comment]
@jameslittle230 Also FYI – `flat.css` looked a little strange, e.g. the oval button, (I'm not sure what it's supposed to look like though). Will go with `basic.css` for now :)

Screenshot 2021-12-12 at 10 17 57

**Pull request merged**